portablemc/cli/__init__.py

```diff
@@ -855,6 +878,11 @@ def forge_resolve(e: ForgeResolveEvent) -> None:
             DownloadStartEvent: self.download_start,
             DownloadProgressEvent: self.download_progress,
             DownloadCompleteEvent: self.download_complete,
+            # Optifine install logic is particular: it needs to be done after the download of the client and installer jar
+            # There is no particular event for the fetching of the version, because the entire json is generated after patch
+            optifine.OptifineStartInstallEvent: lambda e: progress_task("start.optifine.install"),
+            optifine.OptifineEndInstallEvent: lambda e: finish_task("start.optifine.installed", version=e.version),
+            optifine.OptifinePatchEvent: lambda e: progress_task("start.optifine.patching", progress=f"{e.done}/{e.total}"),
```

Comment:
Really good! You can omit the `optifine.` prefix if you directly import the events, as commented on the top of the file.